File: dayu/MaximumSubarray.java
public class MaximumSubarray {
    public int maximumSubArray(int[] nums) {
        if (nums == null) return 0;
        int max = nums[0];
        int sum = max;
        for (int i = 1; i < nums.length; i++) {
            sum = Math.max(nums[i], nums[i] + sum);
            max = (max > sum) ? max : sum;
        }
        return max;
    }
}

File: Sen/Maximum_Subarray.py
class Solution(object):
    def maxSubArray(self, nums): # Greedy, DP
        """
        :type nums: List[int]
        :rtype: int
        """
        maxVal = max(nums)
        current = 0
        for a in nums:
            current += a
            maxVal = max(maxVal, current)
            if current < 0:
                current = 0
        return maxVal

    def maxSubArray(self, nums): # DP
    	if not nums:
    		return 0
    	curSum = maxSum = nums[0]
    	for num in nums[1:]:
    		curSum = max(num, curSum + num)
    		maxSum = max(maxSum, curSum)
    	return maxSum

    def maxSubArray(self, nums): # Divide and Conquer





